server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/storage/ProjectServerIssueStore.java

```diff
@@ -57,30 +58,49 @@ public interface ProjectServerIssueStore {
    *  - existing issues are updated
    *  - closed issues are removed from the store
    */
-  void mergeIssues(String branchName, List<ServerIssue> issuesToMerge, Set<String> closedIssueKeysToDelete, Instant syncTimestamp);
+  void mergeIssues(String branchName, List<ServerIssue> issuesToMerge, Set<String> closedIssueKeysToDelete, Instant syncTimestam, Set<Language> enabledLanguages);
 
   /**
    * Merge provided taint issues to stored ones for the given project:
    *  - new issues are added
    *  - existing issues are updated
    *  - closed issues are removed from the store
    */
-  void mergeTaintIssues(String branchName, List<ServerTaintIssue> issuesToMerge, Set<String> closedIssueKeysToDelete, Instant syncTimestamp);
+  void mergeTaintIssues(String branchName, List<ServerTaintIssue> issuesToMerge, Set<String> closedIssueKeysToDelete, Instant syncTimestamp, Set<Language> enabledLanguages);
 
   /**
    * Merge provided hotspots to stored ones for the given project:
    *  - new hotspots are added
    *  - existing hotspots are updated
    *  - closed hotspots are removed from the store
    */
-  void mergeHotspots(String branchName, List<ServerHotspot> hotspotsToMerge, Set<String> closedHotspotKeysToDelete, Instant syncTimestamp);
+  void mergeHotspots(String branchName, List<ServerHotspot> hotspotsToMerge, Set<String> closedHotspotKeysToDelete, Instant syncTimestamp, Set<Language> enabledLanguages);
 
   /**
    * Return the timestamp of the last issue sync for a given branch.
    * @return empty if the issues of the branch have never been pulled
    */
   Optional<Instant> getLastIssueSyncTimestamp(String branchName);
 
+  /**
+   * Return the last enabled languages of the last issue sync for a given branch.
+   *
+   * @return empty if the issues of the branch have never been pulled
+   */
+  Set<String> getLastIssueEnabledLanguages(String branchName);
```

Comment:
As mentioned above, move this to `Set<Language>` to be consistent, also for `getLastTaintEnabledLanguages` and `getLastHotspotEnabledLanguages`